xarray/tests/test_indexing.py

```diff
@@ -823,3 +830,69 @@ def test_indexing_1d_object_array() -> None:
     expected = DataArray(expected_data)
 
     assert [actual.data.item()] == [expected.data.item()]
+
+
+@requires_dask
+def test_indexing_dask_array():
+    import dask.array
+
+    da = DataArray(
+        np.ones(10 * 3 * 3).reshape((10, 3, 3)),
+        dims=("time", "x", "y"),
+    ).chunk(dict(time=-1, x=1, y=1))
+    with raise_if_dask_computes():
+        actual = da.isel(time=dask.array.from_array([9], chunks=(1,)))
+    expected = da.isel(time=[9])
+    assert_identical(actual, expected)
+
+
+@requires_dask
+def test_indexing_dask_array_scalar():
+    # GH4276
+    import dask.array
+
+    a = dask.array.from_array(np.linspace(0.0, 1.0))
+    da = DataArray(a, dims="x")
+    x_selector = da.argmax(dim=...)
+    actual = da.isel(x_selector)
+    expected = da.isel(x=-1)
+    assert_identical(actual, expected)
+
+
+@pytest.mark.xfail
+@requires_dask
+def test_vectorized_indexing_dask_array():
+    # https://github.com/pydata/xarray/issues/2511#issuecomment-563330352
```

Comment:
I don't know if this one is worth fixing. We'd have to convert Variables and IndexVariables to dask backed Variables and then index with a dask array.

Reply:
We can't fix this. `dask_array.vindex[dask_array]` is unsupported.

https://github.com/dask/dask/issues/8958